test/integration/fileListing.test.js

```diff
@@ -0,0 +1,89 @@
+var assert = require('assert');
+var _ = require('underscore');
+
+var FileListingPage = require('./pageobjects/fileListing.page');
+
+var FEDERALIST_TEST_USER = process.env.FEDERALIST_TEST_USER;
+var fileListingPage;
+
+describe('generic repository file listing tests', function () {
+  this.timeout(15000);
+
+  before(function () {
+    var engine = sails.hooks[sails.config.build.engine];
+    engine.jekyll = function (model, done) {
+      done();
+    };
+    fileListingPage = new FileListingPage(helpers.webdriver.createDriver());
+
+    return fileListingPage.init();
+  });
+
+  after(function () {
+    return fileListingPage.end();
+  });
+
+  it('logs in', function () {
+    return fileListingPage.login();
+  });
+
+  it('opens', function () {
+    return fileListingPage.open();
+  });
+
+  describe('for a repository', function () {
+    var site;
+    before(function (done) {
+      User.find({ username: FEDERALIST_TEST_USER }).exec(function (err, u) {
+        var data = {
+          owner: FEDERALIST_TEST_USER,
+          repository: 'microsite-template',
+          users: [u[0].id]
+        };
+
+        Site.create(data).exec(function (err, s) {
+          if (err) throw err;
+          site = s;
+          done();
+        });
+      });
```

Comment:
How do you feel about promises?

The docs are very brief on the topic, but Waterline and the Sails models use promises too. ([Bluebird](http://bluebirdjs.com/docs/getting-started.html) under the hood).

```javascript
before(function () {
  return User.find({...})
    .then(function (user) {
      // ...
      return Site.create();
    })
    .then(function (s) {
      site = s;
    });
});